infra/terraform/test-org/ci-comment-build-trigger-function/function_source/main.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import sys
+import os
+import json
+import logging
+import requests
+
+from google.cloud.devtools.cloudbuild_v1 import CloudBuildClient as cloudbuild
+from google.cloud.devtools.cloudbuild_v1.types import BuildStep, Build, BuildOptions
+
+CFT_TOOLS_DEFAULT_IMAGE = 'gcr.io/cloud-foundation-cicd/cft/developer-tools'
+CFT_TOOLS_DEFAULT_IMAGE_VERSION = '0.11.0'
+ENABLED_MODULES = [
+    'terraform-google-cloud-storage',
+]
+
+
+def main(event, context):
+    """ Triggers a new downstream build based on a PubSub message originating from a parent cloudbuild """
+    # if cloud build project is not set, exit
+    if not os.getenv('CLOUDBUILD_PROJECT'):
+        logging.warn('Cloud Build project not set')
+        sys.exit(1)
+    # if no data in PubSub event, log and exit
+    if 'data' not in event:
+        logging.info('Unable to find data in PubSub event')
+        sys.exit(1)
+    # decode data in PubSub event
+    data = json.loads(base64.b64decode(event['data']).decode('utf-8'))
+    # if the parent build originated from CF, ignore
+    if data['substitutions'].get('_IS_TRIGGERED_BY_CF', False):
+        logging.warn('Triggered by CF, Ignoring')
+        return
+    logging.info('Parent build not triggered by CF')
+    # if parent build is not a lint build, ignore
+    if 'lint' not in data['tags']:
+        logging.warn('Parent build is not a lint build')
+        return
+    # if parent build has not started, or is in any other state, ignore
+    if data['status'] != 'WORKING':
+        logging.warn('Parent build is not in WORKING status')
+        return
+    logging.info('Parent build is in WORKING status')
+    # if repo ref for the parent build has not enabled PR bot, ignore
+    if data['substitutions']['REPO_NAME'] not in ENABLED_MODULES:
+        logging.warn('Not a supported repo')
+        return
+    if data['substitutions'].get('_DOCKER_TAG_VERSION_DEVELOPER_TOOLS', False):
+        logging.info(
+            f'Found _DOCKER_TAG_VERSION_DEVELOPER_TOOLS. Setting tools image version to {data["substitutions"]["_DOCKER_TAG_VERSION_DEVELOPER_TOOLS"]}'
+        )
+        CFT_TOOLS_DEFAULT_IMAGE_VERSION = data['substitutions'][
+            '_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+        ]
+    # Cloud Build seems to have a bug where if a build is re run thought Github UI, it will not set _PR_NUMBER or _HEAD_REPO_URL
```

Comment:
nit.

```suggestion
    # Cloud Build seems to have a bug where if a build is re run through Github UI, it will not set _PR_NUMBER or _HEAD_REPO_URL
```